src/software/thunderscope/replay/proto_player.py

```diff
@@ -0,0 +1,334 @@
+import time
+import threading
+import base64
+import os
+import logging
+import gzip
+import glob
+import proto
+from proto.import_all_protos import *
+from extlibs.er_force_sim.src.protobuf.world_pb2 import *
+from software.thunderscope.replay.replay_constants import *
+
+
+class ProtoPlayer(object):
+
+    """Plays back a proto log folder. All the playback is handled by a worker
+    thread running in the background.
+        
+        
+                             current_chunk
+                                  │
+                                  │
+                       ┌─────┐ ┌──▼──┐            ┌─────┐
+                       │CHUNK│ │CHUNK│            │CHUNK│
+                    ┌──┼─────┼─┼─►   │ ────────── │     │
+                    │  │     │ │     │            │     │
+                    │  │     │ │     │            │     │
+                    │  └─────┘ └─────┘            └─────┘
+                    │     0       1                  N
+                    │             ▲
+                    │             └──current_chunk_index
+                    │
+             current_entry_index
+        
+    The player will load chunks in order and play them back at the given playback
+    speed. If the seek function is called with a specific time, the player will
+    update the 3 variables (shown above) to point to the chunk and entry (in the
+    chunk) that contains the data at that time and continue playing from there.
+
+    """
+
+    def __init__(self, log_folder_path, proto_unix_io):
+        """Creates a proto player that plays back all protos
+
+        :param log_folder_path: The path to the log file.
+        :param proto_unix_io: The proto_unix_io to send the protos to.
+            
+        """
+        self.log_folder_path = log_folder_path
+        self.proto_unix_io = proto_unix_io
+        self.current_packet_time = 0.0
+        self.end_time = 0
+
+        # Don't continue setting things up if the log folder was not provided
+        if self.log_folder_path is None:
+            return
+
+        # Playback controls, see __play_protobufs
+        self.stop_playing = False
```

Comment:
any reason why we use stop_playing as a name instead of is_playing

Reply:
no reason, flipped to is_playing because its works better anyway